### PR TITLE
MAGECLOUD-2514: Add List of Modules which were Enabled by module:refresh Command to cloud.log

### PR DESCRIPTION
**Pull request by oshmyheliuk**
### Description
Added more logging to `module:refresh` command about the list of enabled modules.

### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-2514

### Manual testing scenarios

https://jira.corp.magento.com/browse/MAGECLOUD-43
https://jira.corp.magento.com/browse/MAGECLOUD-94

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Config/Module.php [hunk 1]
```diff
@@ -36,22 +36,40 @@ public function __construct(ConfigInterface $config, ShellInterface $shell)
...
+
         if (!$moduleConfig) {
-            $this->shell->execute('php ./bin/magento module:enable --all --ansi --no-interaction');
             $this->config->reset();
+        } else {
+            $this->config->update(['modules' => $moduleConfig]);
+        }
 
-            return;
+        return $this->parseModuleList($process->getOutput());
```
Comment (billygilbert) [inline src/Config/Module.php]: This is not working as expected.  This shows modules that were previously disabled as being enabled, though they remain disabled after.  Steps to recreate:

1. `./vendor/bin/ece-tools module:refresh`
2. Modify `app/etc/config.php`, disable `Temando_Shipping` and `Vertex_Tax` by setting to `0`
3. `./vendor/bin/ece-tools module:refresh`

Expected result:
`No modules were changed.`

Actual result:
```INFO: The following modules have been enabled:
Temando_Shipping
Vertex_Tax
```
Modules are disabled in `app/etc/config.php` still

**Review by billygilbert — CHANGES_REQUESTED** [review]

File: src/Config/Module.php (hunk 1 above)
Comment (oshmyheliuk) [reply src/Config/Module.php]: Thanks for your comment. Implementation was fixed.

**Review by billygilbert — DISMISSED** [review]
Looks great!

File: src/Config/Module.php [hunk 2]
```diff
@@ -36,22 +36,25 @@ public function __construct(ConfigInterface $config, ShellInterface $shell)
 
     /**
      * Reconciling installed modules with shared config.
+     * Returns list of enabled modules or an empty array if no modules were enabled.
```
Comment (NadiyaS) [inline src/Config/Module.php]: I would say that it returns "new enabled/added modules".

**Review by NadiyaS — DISMISSED** [review]

**Review by billygilbert — DISMISSED** [review]

**Comment by mveeramneni** [comment]
QA Done for current implementation, but waiting for a response from Yevhenii on the ticket.

**Review by NadiyaS — DISMISSED** [review]

File: src/Command/ModuleRefresh.php [hunk 3]
```diff
@@ -58,7 +58,12 @@ public function execute(InputInterface $input, OutputInterface $output)
     {
         try {
             $this->logger->info('Refreshing modules started.');
-            $this->config->refresh();
+            $enabledModules = $this->config->refresh();
+            $this->logger->info(
+                $enabledModules ?
+                    'The following modules have been enabled:' . PHP_EOL . implode(PHP_EOL, $enabledModules) :
+                    'No modules were changed.'
+            );
```
Comment (billygilbert) [inline src/Command/ModuleRefresh.php]: This code is duplicated in src/Process/Build/RefreshModules.php. We should refactor it into src/Config/Module.php

**Review by billygilbert — CHANGES_REQUESTED** [review]

File: src/Command/ModuleRefresh.php (hunk 3 above)
Comment (oshmyheliuk) [reply src/Command/ModuleRefresh.php]: It's not a responsibility of `src/Config/Module.php` class to write into log files.
I'll refactor it by using RefreshModules process in ModuleRefresh command.

Comment (billygilbert) [reply src/Command/ModuleRefresh.php]: I'm fine with that solution.

**Comment by mveeramneni** [comment]
Waiting for @YPyltiai response on ticket https://magento2.atlassian.net/browse/MAGECLOUD-2514

**Review by NadiyaS — APPROVED** [review]

**Review by billygilbert — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA passed

**Pull request merged**